### PR TITLE
Leyli/search name

### PR DESCRIPTION
**Pull request by leyliramazanova**

### CODE REVIEW

File: backend/src/convert-search-name.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const convertSearchName = ({ name }) => {
+  const params = convertSearchNameParams(name)
+  const query = convertSearchNameQuery
+  return { params, query }
+}
+
+const convertSearchNameParams = (name) => {
+  const params = { person: name }
```
Comment (kwannoel) [inline backend/src/convert-search-name.js:8]: property name should be specific

use `{'name' : name}` instead.

Comment (leyliramazanova) [reply backend/src/convert-search-name.js:8]: hm, then there's literally no point in having this function?

Comment (leyliramazanova) [reply backend/src/convert-search-name.js:8]: since we are using substitution in queries, why don't we just directy sub in the name?

File: backend/src/convert_search_name_response.unit.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    2,
+    [new Integer(6, 4), 'James'],
+    { 'id(person)': 0, 'person.value': 1 })
+  ]
+}
+
+const convertedResponse = {
+  matches: [{
+    name: 'James',
+    id: 0
```
Comment (kwannoel) [inline backend/src/convert_search_name_response.unit.test.js:38]: details as well. See the specs https://www.notion.so/holmusk/Pipeline-specifications-e8f3648483404c04aaf04a69bfdccb1c#f8e48e465af840d18508a019121cba63

File: backend/src/process-resource.unit.test.js [hunk 3]
```diff
@@ -87,6 +82,24 @@ test('Should throw missing fields error', () => {
...
 })
 
+const invalidName = {
+  name: 'James T.'
+}
+
+test('Should throw error for an invalid name', () => {
+  expect(() => {
+    process.searchName(invalidName)
+  }).toThrow()
```
Comment (kwannoel) [inline backend/src/process-resource.unit.test.js:92]: Since we expect a specific error to be thrown,
you should include the error class here. Otherwise, the test will pass as well if other errors were thrown.

See this for example: https://github.com/Holmusk/HolmuskDB/pull/97/files#diff-781bbc3775153d9bb558d2dec9ad2c2bR39

File: backend/tests/server.int.test.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ const request = require('supertest')
 const db = require('./utils/db.js')
 const env = require('../src/env')
 const { runDriver } = require('../src/db-utils')
-const { runServer } = require('../src/server')
+const { runServer } = require('../src/server.js')
```
Comment (kwannoel) [inline backend/tests/server.int.test.js:7]: Why is there `.js` here

File: backend/tests/server.int.test.js [hunk 5]
```diff
@@ -22,20 +23,11 @@ describe('Testing endpoints', () => {
 
   // End driver, server
   afterAll(async () => {
+    await db.teardown(driver)
     await driver.close()
     await serverHandle.close()
   })
 
-  // Run each test against clean db
```
Comment (kwannoel) [inline backend/tests/server.int.test.js:29]: why are these removed

File: backend/tests/server.int.test.js [hunk 6]
```diff
@@ -14,6 +14,7 @@ describe('Testing endpoints', () => {
   beforeAll(async () => {
     env.init()
     driver = await runDriver()
+    await db.teardown(driver)
```
Comment (kwannoel) [inline backend/tests/server.int.test.js:17]: These shouldn't be added. The `beforeEach` and `afterEach` hooks handle the teardown step for each test.

File: backend/src/convert-search-name.js (hunk 1 above)
Comment (kwannoel) [reply backend/src/convert-search-name.js:3]: As discussed we can do so, let's do that.

File: backend/tests/server.int.test.js [hunk 7]
```diff
@@ -276,4 +276,55 @@ describe('Testing endpoints', () => {
     expect(res.status).toBe(500)
     done()
   })
+
+  // Search person
+  test('Should give a valid response for /db/search/name', async (done) => {
+    const addResource1 = {
```
Comment (kwannoel) [inline backend/tests/server.int.test.js:282]: Let's remove the test dependency on add resource, write our own vanilla query to populate the database and use `runQuery` to run the query.

**Comment by kwannoel** [comment]
need to rebase

File: backend/src/process-resource.unit.test.js (hunk 3 above)
Comment (leyliramazanova) [reply backend/src/process-resource.unit.test.js:92]: will be included in the next commit. Thought putting it in the server int tests would be enough.

File: backend/tests/server.int.test.js (hunk 7 above)
Comment (kwannoel) [reply backend/tests/server.int.test.js:282]: To clarify, vanilla query just means a native query in neo4j.

e.g.
```
CREATE (person:Person {value: 'James'})
```
